docs/src/main/asciidoc/ap4k.adoc

```diff
@@ -0,0 +1,155 @@
+= {project-name} - Deploying an Application to Kubernetes and OpenShift
+
+This guide covers:
+
+* The deployment of the application to Kubernetes
+* The deployment of the application to OpenShift
+
+== Prerequisites
+
+To complete this guide, you need:
+
+* roughly 15 minutes
+* an IDE
+* JDK 1.8+ installed with `JAVA_HOME` configured appropriately
+* Apache Maven 3.5.3+
+* access to Kubernetes or Openshift cluster (Minikube and Minishift are both viable options)
+* have the following binaries on your PATH: `docker` (for building and pushing images), `kubectl` (for deploying to Kubernetes) and `oc` (for deploying to Openshift)
+
+Remember, you need to configure Maven as indicated in the link:maven-config.html[Maven configuration page].
+
+== Creating the Maven project
+
+First, we need a new project that contains the ap4k extension. This can be done using the following command:
+
+[source, subs=attributes+]
+----
+mvn io.quarkus:quarkus-maven-plugin:${quarkus-version}:create \
+    -DprojectGroupId=org.acme \
+    -DprojectArtifactId=ap4k \
+    -DclassName="org.acme.rest.GreetingResource" \
+    -Dpath="/greeting" \
+    -Dextensions="ap4k"
+----
+
+With project setup, proceed to create a `src/main/java/org/acme/rest/MyApplication.java` file that contains the following source:
+
+[source,java]
+----
+import javax.ws.rs.core.Application;
+
+public class MyApplication extends Application {
+
+}
+----
+
+This is the class we will use in the following sections to add our ap4k annotations.
+
+== Enable Kubernetes support
+
+By simply adding the `io.ap4k.kubernetes.annotation.KubernetesApplication` annotation to the `MyApplication` class, you will now be able to create a `Kubernetes` resources file
+that can later by applied to to deploy the application to Kubernetes.
+
+`MyApplication` should look like so:
+
+[source,java]
+----
+import javax.ws.rs.core.Application;
+import io.ap4k.kubernetes.annotation.KubernetesApplication;
+import io.ap4k.kubernetes.annotation.Port;
+
+@KubernetesApplication(ports = @Port(name = "http", containerPort = 8080), group = "yourDockerUsername")
+public class MyApplication extends Application {
+
+}
+----
+
+Now, when you build the project using `mvn package`, amongst the other file that are created you should see two files named `kubernetes.json` and `kubernetes.yaml`
+in `target/wiring-classes/META-INF/ap4k/` directory.
```

Comment:
s/in/in the/
I'm not sure it's the right directory to create your descriptor files. It's supposed to contain some generated classes.

Reply:
Indeed, and that is how I initially had set things up, but after a review by Stuart I changed it. Basically now Quarkus is the one that actually writes the file to disc, so ends it up there (I initially had it so ap4k itself wrote to the disc)